thorlcr/activities/thdiskbaseslave.cpp

```diff
@@ -235,13 +234,36 @@ void CDiskReadSlaveActivityBase::init(MemoryBuffer &data, MemoryBuffer &slaveDat
     {
         deserializePartFileDescriptors(data, partDescs);
 
-        // put temp files in individual slave temp dirs (incl port)
-        if ((helper->getFlags() & TDXtemporary) && (!container.queryJob().queryUseCheckpoints()))
-            partDescs.item(0).queryOwner().setDefaultDir(queryTempDir());
+        if (helper->getFlags() & TDXtemporary)
+        {
+            // put temp files in individual slave temp dirs (incl port)
+            if (!container.queryJob().queryUseCheckpoints())
+                partDescs.item(0).queryOwner().setDefaultDir(queryTempDir());
+        }
+        else
+        {
+            ISuperFileDescriptor *super = partDescs.item(0).queryOwner().querySuperFileDescriptor();
+            if (super)
+            {
+                unsigned numSubFiles = super->querySubFiles();
+                for (unsigned i=0; i<numSubFiles; i++)
+                    subFileStats.push_back(new CRuntimeStatisticCollection(diskReadRemoteStatistics));
```

Comment:
Each worker may only be handling parts from some sub files, e.g. a super with 2 subs of 2 parts, being read on a 4-way. Meaning this would serialize metrics for subs that aren't being read by this slave.

But not a big issue and don't think should change, just an observation.

Reply:
If so I think that will end up serializing a single 0 byte, so trivial overhead.